substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -2473,6 +2513,844 @@ static boolean not(byte b) {
     static boolean not(int i) {
         return !(CTypeConversion.toBoolean(i));
     }
+
+
+    /* Do not re-format commented out code: @formatter:off */
+    //  1878  /*
+    //  1879   * mcast_join_leave: Join or leave a multicast group.
+    //  1880   *
+    //  1881   * For IPv4 sockets use IP_ADD_MEMBERSHIP/IP_DROP_MEMBERSHIP socket option
+    //  1882   * to join/leave multicast group.
+    //  1883   *
+    //  1884   * For IPv6 sockets use IPV6_ADD_MEMBERSHIP/IPV6_DROP_MEMBERSHIP socket option
+    //  1885   * to join/leave multicast group. If multicast group is an IPv4 address then
+    //  1886   * an IPv4-mapped address is used.
+    //  1887   *
+    //  1888   * On Linux with IPv6 if we wish to join/leave an IPv4 multicast group then
+    //  1889   * we must use the IPv4 socket options. This is because the IPv6 socket options
+    //  1890   * don't support IPv4-mapped addresses. This is true as per 2.2.19 and 2.4.7
+    //  1891   * kernel releases. In the future it's possible that IP_ADD_MEMBERSHIP
+    //  1892   * will be updated to return ENOPROTOOPT if uses with an IPv6 socket (Solaris
+    //  1893   * already does this). Thus to cater for this we first try with the IPv4
+    //  1894   * socket options and if they fail we use the IPv6 socket options. This
+    //  1895   * seems a reasonable failsafe solution.
+    //  1896   */
+    //  1897  static void mcast_join_leave(JNIEnv *env, jobject this,
+    //  1898                               jobject iaObj, jobject niObj,
+    //  1899                               jboolean join) {
+    static void mcast_join_leave(Target_java_net_PlainDatagramSocketImpl self, InetAddress iaObj, NetworkInterface niObj, boolean join) throws IOException {
+        // The Linux code path uses a different mreq structure, so for type safety we have to factor it out completely.
+        if (IsDefined.__linux__()) {
+            //  1901      jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
+            final FileDescriptor fdObj = self.fd;
+            //  1902      jint fd;
+            int fd;
```

Comment:
This leads to a style check error: [The local variable fd is hiding a field from type Target_java_net_PlainDatagramSocketImpl](https://travis-ci.org/oracle/graal/jobs/548381854#L2676). I would rename the field to something like `fdObject` because there are multiple locals with the name `fd`.

Reply:
This kind of check is subjective so I'm unlikely to guess that it exists; is there any way to run the style check process locally?

Reply:
You should be able to run with [`mx gate --strict-mode --tags style`](https://travis-ci.org/oracle/graal/jobs/548381854#L2783), but you need to have the right versions of Eclipse JDT, pylint, checkstyle installed.